### PR TITLE
feat(api): Add G-Code parsing for Thermocycler

### PR DESCRIPTION
**Pull request by DerekMaggio**
# Overview

Add G-Code parsing for Thermocycler and tests for said parsing.

# Changelog

* Add G-Code functionality defs for Thermocycler G-Code functions
* Add tests

# Review requests

None

# Risk assessment

None

### CODE REVIEW

File: api/src/opentrons/hardware_control/g_code_parsing/g_code_functionality_defs/thermocycler/edit_pid_params_g_code_functionality_def.py [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+    def _generate_command_explanation(cls, g_code_args: Dict[str, str]) -> str:
+        message_list = []
+        for arg in cls.EXPECTED_ARGS:
+            g_code_arg_val = g_code_args.get(arg)
+            if g_code_arg_val is not None:
+                message_temp = Template(cls.ValDefinedMessage[arg].value)
+                message = message_temp.substitute(val=g_code_arg_val)
+                message_list.append(message)
+
+        return 'Editing PID values to the following:' + ''.join(message_list)
```
Comment (amitlissack) [inline api/src/opentrons/hardware_control/g_code_parsing/g_code_functionality_defs/thermocycler/edit_pid_params_g_code_functionality_def.py]: Minor nit: We tend to use f"" string formatting for example....
f"Editing.....: {''.join(message_list)}"

File: api/tests/opentrons/hardware_control/g_code_parsing/test_g_code.py [hunk 2]
```diff
@@ -236,9 +237,9 @@ def magdeck_g_codes() -> List[GCode]:
 
     for g_code in g_code_list:
         if len(g_code) > 1:
-            magdeck_g_codes = ', '.join([code.g_code for code in g_code])
+            g_codes = ', '.join([code.g_code for code in g_code])
```
Comment (amitlissack) [inline api/tests/opentrons/hardware_control/g_code_parsing/test_g_code.py]: Minor nit: No need for [ ] in the join.  Can do `join(code.g_code for code in g_code)`

**Review by amitlissack — APPROVED** [review]
Good stuff! Two minor nits.

File: api/tests/opentrons/hardware_control/g_code_parsing/test_g_code.py (hunk 2 above)
Comment (DerekMaggio) [reply api/tests/opentrons/hardware_control/g_code_parsing/test_g_code.py]: Addressed

File: api/src/opentrons/hardware_control/g_code_parsing/g_code_functionality_defs/thermocycler/edit_pid_params_g_code_functionality_def.py (hunk 1 above)
Comment (DerekMaggio) [reply api/src/opentrons/hardware_control/g_code_parsing/g_code_functionality_defs/thermocycler/edit_pid_params_g_code_functionality_def.py]: Addressed

**Pull request merged**